src/codeMarker.ts

```diff
@@ -551,6 +563,66 @@ export class CodeMarker implements vscode.TreeDataProvider<TreeEntry> {
         this.refresh(uri);
     }
 
+    addPartialAudited(): void {
+        const editor = vscode.window.activeTextEditor;
+        if (editor === undefined) {
+            return;
+        }
+        const uri = editor.document.uri;
+        // get path relative to workspace
+        const relativePath = path.relative(this.workspacePath, uri.fsPath);
+
+        // check if file is already in list
+        const index = this.auditedFiles.findIndex((file) => file.path === relativePath);
+
+        // if file is already audited ignore
+        if (index > -1) {
+            return;
+        }
+
+        const location = this.getActiveSelectionLocation();
+        const exactMatchIndex = this.partialAuditedFiles.findIndex(
+            (file) => file.path === relativePath && file.location.startLine === location.startLine && file.location.endLine === location.endLine,
+        );
+
+        // this section is already marked. Remove it then
+        if (exactMatchIndex > -1) {
+            this.partialAuditedFiles.splice(exactMatchIndex, 1);
+        } else {
+            // allows to remove sections from the begining or end
+            const partialMatchIndex = this.partialAuditedFiles.findIndex(
+                (file) => file.path === relativePath && (file.location.startLine === location.startLine || file.location.endLine === location.endLine),
+            );
+            if (partialMatchIndex > -1) {
+                const partialMatch = this.partialAuditedFiles[partialMatchIndex];
+                if (partialMatch.location.startLine === location.startLine) {
+                    partialMatch.location.startLine = location.endLine + 1;
+                }
+                if (partialMatch.location.endLine === location.endLine) {
+                    partialMatch.location.endLine = location.startLine - 1;
+                }
```

Comment:
I guess the behavior unclarity would be resolved by having the behavior you suggested. I had to add it because there was no way to remove partially audited entries without marking the whole file as audited and then unaudited. This but would also delete all previous entries

Reply:
so new behavior is now in there:
if there is an entry as an exact match => remove it
if there is an entry as a partial match => add it
if there is an entry that is around the selected area => splice it (if necessary)